Integrating Scriber into Corgie

supersergiy opened the pull request:
Integrating the basic version of scriber into `corgie` and `mazepa`.

Verified to produce correct results with `NormalizationJob` and `AlignBlockJob`.

MergeRender tasks shrunk to 2KB


**To test**: use `mazepa` -e git+https://github.com/supersergiy/scheduler.git@8905783b680cafcebfec52fec7f0e68e4be0e5fd#egg=mazepa`

supersergiy committed:
scriber tested with apply processor

supersergiy committed:
scriber

nkemnitz reviewed (changes requested):
* `objectscriber` needs to be added to the requirements
*  This line in `MiplessCloudVolume.__init__` is problematic: https://github.com/seung-lab/corgie/blob/d6a973a575139b27302a63c1f30555d6617d0a66/corgie/mipless_cloudvolume.py#L91-L92 Only the scheduler should create/overwrite the info file - right now all workers will try to rewrite the info file simultaneously, which is causing e.g. GCS to panic with 429 errors.

Is there a good way to figure out if the job was initialized by the worker or the scheduler?

nkemnitz commented:
* DownsamplingJob workers suddenly complained about the `dst_layer` (not specified) being readonly

supersergiy commented:
Thanks for the feedback! I'll address these issues asap

supersergiy committed:
addressed nicos issues

supersergiy commented:
Added a global `settings.py` file which keeps track of whether the process is worker or not. Workers don't upload `info`.
Added `objectscriber` to `setup.py`.
Solved the downsample issues.
@nkemnitz

nkemnitz commented on corgie/cli/downsample_by_spec.py:
```diff
@@ -71,20 +71,22 @@ def downsample_by_spec(
-    )
-
-    with open(spec_path, "r") as f:
+        with open(spec_path, "r") as f:
```
```suggestion
    with open(spec_path, "r") as f:
```

nkemnitz reviewed (changes requested):
Just one typo - see suggestion
Besides objectscriber, does this PR also require a minimal mazepa version? If so, that should be specified.

supersergiy committed:
Update corgie/cli/downsample_by_spec.py

Co-authored-by: Nico Kemnitz <nkemnitz@princeton.edu>

supersergiy committed:
m
